File: 1_two_sum/1.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        hash_map = {}
        for index, num in enumerate(nums):
            complement = target - num
            if complement in hash_map:
                return [hash_map[complement], index]
            hash_map[num] = index
        return None

# Solutionクラスのインスタンスを作成
solution = Solution()

# テストケースの実行と出力
print(solution.twoSum([2, 7, 11, 15], 9))  # => [0, 1]
print(solution.twoSum([3, 2, 4], 6))       # => [1, 2]
print(solution.twoSum([3, 3], 6))          # => [0, 1]

File: 1/1.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}

def two_sum(nums, target)
    hash = {}
    nums.each_with_index do |num, index|
        complement = target - num
        if hash.has_key?(complement)
            return [hash[complement], index]
        end
        hash[num] = index
    end
    nil
end

puts two_sum([2, 7, 11, 15], 9).inspect # => [0, 1]
puts two_sum([3, 2, 4], 6).inspect      # => [1, 2]
puts two_sum([3, 3], 6).inspect         # => [0, 1]